### ISSUE TITLE
PublishMessageAsync

### ISSUE THREAD

**Issue by koderi-dp** [loekd/ServiceFabric.PubSubActors#43]
i saw something that could be bug,

`PublishMessageAsync(StatelessService service, object message, Uri brokerServiceName = null) `

receives object message, then if brokerServiceName is not null

`   var brokerService = await _brokerServiceLocator.GetBrokerServiceForMessageAsync(message, brokerServiceName);`

it is calling 

```
public Task<IBrokerService> GetBrokerServiceForMessageAsync(object message, Uri brokerServiceName)
		{
			return GetBrokerServiceForMessageAsync(message.GetType().FullName, brokerServiceName);
		}
```
which is calling

```
 var resolvedPartition = await GetPartitionForMessageAsync(messageTypeName, brokerServiceName);

```
GetPartitionForMessage message parameter is object, so it's receiving string from above call
```
Task<ServicePartitionKey> GetPartitionForMessageAsync(object message, Uri brokerServiceName)
```

then in that function you are 
```
string messageTypeName = (message.GetType().FullName);

```

which will give you System.String, so index will in most cases be same value

```
int index = Math.Abs(messageTypeName.GetHashCode() % _cachedPartitions.Count);
			var partition = _cachedPartitions[index];
```

maybe just remove 

`Task<IBrokerService> GetBrokerServiceForMessageAsync(string messageTypeName, Uri brokerServiceName);`

**Comment by loekd** [comment]
Hi again @danijel-peric, thanks for reporting this.
It's fixed in v7.3.6
Please report back if it works for you.

**Comment by koderi-dp** [comment]
Working
